app/services/course/assessment/programming_evaluation_service.rb

```diff
@@ -105,7 +105,7 @@ def initialize(language, memory_limit, time_limit, max_time_limit, package, time
     @memory_limit = memory_limit || MEMORY_LIMIT
     @time_limit = time_limit ? [time_limit, max_time_limit].min : max_time_limit
     @package = package
-    @timeout = timeout || DEFAULT_TIMEOUT
+    @timeout = timeout || [DEFAULT_TIMEOUT, time_limit].max
```

Comment:
The DEFAULT_TIMEOUT is in minutes but time_limit is in seconds. Also, I think it is better to take directly from `@time_limit` as it is the computed time_limit for the evaluation. To be more defensive, it is also good to check if `@time_limit` is undefined or not

Reply:
Updated to `[DEFAULT_TIMEOUT.to_i, @time_limit.to_i].max` to explicitly compare using integers instead of time objects.